src/notification.rs

```diff
@@ -82,25 +82,40 @@ impl Notification {
         }
     }
 
-    pub fn get_notification(&self) -> Option<NotificationCode> {
-        let result = match self.get_type() {
-            NotificationType::Type_Notification | NotificationType::Type_ControllerCommand =>
-                Some(unsafe { extern_notification::notification_get_notification(self.ptr) }),
+    pub fn get_controller_state(&self) -> Option<ControllerState> {
+        if self.get_type() != NotificationType::Type_ControllerCommand {
+            return None;
+        }
+        match unsafe { extern_notification::notification_get_notification(self.ptr) } {
+            0  => Some(ControllerState::ControllerStateNormal),
+            1  => Some(ControllerState::ControllerStateStarting),
+            2  => Some(ControllerState::ControllerStateCancel),
+            3  => Some(ControllerState::ControllerStateError),
+            4  => Some(ControllerState::ControllerStateWaiting),
+            5  => Some(ControllerState::ControllerStateSleeping),
+            6  => Some(ControllerState::ControllerStateInProgress),
+            7  => Some(ControllerState::ControllerStateCompleted),
+            8  => Some(ControllerState::ControllerStateFailed),
+            9  => Some(ControllerState::ControllerStateNodeOK),
+            10 => Some(ControllerState::ControllerStateNodeFailed),
+            _  => None
```

Comment:
The c_like_enum macro should make it easier to do this: `ControllerState::from_u8(val)` should return Option<ControllerState>`